.github/workflows/sync-upstream.yml

```diff
@@ -24,6 +24,6 @@ jobs:
           update-script: |
             source_url="$(curl -sL https://developer.android.com/studio/index.html | grep -Eo '"((https)?://.*linux.tar.gz)"' | tr -d '"')"
             codename="$(curl -sL https://developer.android.com/studio/index.html | grep -Po 'Download Android Studio (?!today)[A-Za-z0-9]+' | head -n1 | cut -d ' ' -f4)"
-            version="$(echo "$source_url" | rev | cut -d '/' -f2 | rev && echo "$codename")"
-            sed -i 's/^\(source: \).*$/\1'"\"$source_url\""'/' snap/snapcraft.yaml
+            version="$(basename "$source_url" .tar.gz | grep -oP '\d+\.\d+\.\d+\.\d+') $codename"
+            yq -i ".parts.android-studio.source = \"$source_url\"" snap/snapcraft.yaml
```

Comment:
Please revert this change to `yq` - this will mutate and reformat the whole yaml file each time it's run, and not necessarily for the better. `sed` was likely a good choice here.

Reply:
`sed` is not working with `url`s as you can see in this [discussion](https://github.com/snapcrafters/android-studio/pull/118). That's why this PR is here to change it. It's getting confused with the `/` in the `url`s. I tried the same command, and it works if it doesn't contain `/`.

Reply:
You can use a different separator with `sed` which I think will solve it. Something like `s|foo|bar`. I've not tested, but the original would then look something like this:

```suggestion
            sed -i 's|^\(source: \).*$|\1'"\"$source_url\""'|' snap/snapcraft.yaml
```

Reply:
Let me test it then

Reply:
`sed -i 's|^\(source: #).*$|"\"$source_url\"|"' snap/snapcraft.yaml`
`sed -i 's#^\(source: #).*#"\"$source_url\"#"' snap/snapcraft.yaml`
`sed -i 's#^\(source: \).*$#\1'"\"$source_url\""'#' snap/snapcraft.yaml`

tried all three above, didn't work

Reply:
```
sed -i 's|^\(source: \).*$|\1'"\"$source_url\""'|' snap/snapcraft.yaml
sed -i 's#^\(source: \).*$#\1'"\"$source_url\""'#' snap/snapcraft.yaml
sed -i 's#^\(source: #).*#"\"$source_url\"#"' snap/snapcraft.yaml
sed -i 's|^\(source: #).*$|"\"$source_url\"|"' snap/snapcraft.yaml
```
tried all of these, but none worked. :-(

![image](https://github.com/snapcrafters/android-studio/assets/72045785/98154da1-6b89-4006-b8b7-58edb15bf787)

Reply:
The following works:

```
sed -i "s|source: .*$|source: $source_url|g" snap/snapcraft.yaml
```

Reply:
Thanks! Done!